runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/work/refresh/Heartbeats.java

```diff
@@ -45,21 +46,32 @@ public abstract static class Builder {
 
     abstract ImmutableList.Builder<RefreshableWork> workBuilder();
 
-    public final Builder addWork(RefreshableWork work) {
+    public final Builder add(
+        String computationId, RefreshableWork work, DataflowExecutionStateSampler sampler) {
       workBuilder().add(work);
+      addHeartbeatRequest(computationId, createHeartbeatRequest(work, sampler));
       return this;
     }
 
+    private Windmill.HeartbeatRequest createHeartbeatRequest(
+        RefreshableWork work, DataflowExecutionStateSampler sampler) {
+      return Windmill.HeartbeatRequest.newBuilder()
+          .setShardingKey(work.getShardedKey().shardingKey())
+          .setWorkToken(work.id().workToken())
+          .setCacheToken(work.id().cacheToken())
+          .addAllLatencyAttribution(work.getHeartbeatLatencyAttributions(sampler))
+          .build();
+    }
+
     abstract Builder setHeartbeatRequests(
```

Comment:
can this be removed so that add is used?

Reply:
done